lib/ex_unit/lib/ex_unit/callbacks.ex

```diff
@@ -801,9 +812,19 @@ defmodule ExUnit.Callbacks do
     end)
   end
 
-  defp compile_setup_call(callback) do
+  defp compile_setup_call(callback) when is_atom(callback) do
```

Comment:
We need to revert this change. It is not guaranteed to be a literal atom at compile time. It can be any AST. The validation happens later. :)

Reply:
I'm not sure if I follow, I tried some non-literal and it worked. Could you please provide an example?

Reply:
```
atom = :foo
setup atom
```

:)

Reply:
This is what I tried but it didn't cause any issues, pushed for reference.

Reply:
Perfect. it was my bad. I think you can remove the last commit that adds the reference because it is most likely already tested elsewhere. :)

Reply:
Removed the commit